### PR TITLE
add design resource

### PR DESCRIPTION
**Pull request by samundra**
API - Design GET https://github.com/apacig/opendesign/issues/9

API - Design POST https://github.com/apacig/opendesign/issues/10

API - Design PUT https://github.com/apacig/opendesign/issues/11

API - Design DELETE https://github.com/apacig/opendesign/issues/11

API - Design List all Designs

### CODE REVIEW

File: backend/app/Brief.php [hunk 1]
```diff
@@ -54,4 +54,10 @@ public function designs()
     {
         return $this->hasMany('App\Design');
     }
+
+    public function project()
+    {
+        $project = $this->belongsTo('App\Project');
```
Comment (bikalbasnet) [inline backend/app/Brief.php]: can do `return  $this->belongsTo('App\Project')`

File: backend/app/Design.php [hunk 2]
```diff
@@ -38,4 +42,14 @@ public function brief()
...
     }
+
+    public function assets()
+    {
+        return $this->hasMany('App\DesignAsset');
+    }
+
+    public function user()
+    {
+        return $this->belongsTo('App\User')->get()->first();
```
Comment (bikalbasnet) [inline backend/app/Design.php]: ` return $this->belongsTo('App\User')` ??

File: backend/app/Http/Middleware/JsonApiMiddleware.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    * Handle an incoming request.
+    *
+    * @param \Illuminate\Http\Request $request
+    * @param \Closure $next
+    *
+    * @return mixed
+    */
+    public function handle($request, Closure $next)
+    {
+        if (in_array($request->getMethod(), self::PARSED_METHODS)) {
```
Comment (AlexStansfield) [inline backend/app/Http/Middleware/JsonApiMiddleware.php]: Is this needed? Laravel converts json automatically if you set the content-type

File: backend/app/Http/Controllers/DesignController.php [hunk 4]
```diff
@@ -113,76 +114,110 @@ public function getComments($id)
     /**
      * Display a listing of the resource.
      *
-     * @return \Illuminate\Http\Response
+     * @return \Illuminate\Http\JsonResponse
      */
     public function index(Request $request)
```
Comment (bikalbasnet) [inline backend/app/Http/Controllers/DesignController.php]: We can remove `$request` param

File: backend/database/migrations/2017_11_24_140740_create_initial_schema.php [hunk 5]
```diff
@@ -72,11 +72,11 @@ public function up()
 
         Schema::create('design_asset', function (Blueprint $table) {
             $table->increments('id');
-            $table->integer('brief_id')->unsigned();
+            $table->integer('design_id')->unsigned();
```
Comment (AlexStansfield) [inline backend/database/migrations/2017_11_24_140740_create_initial_schema.php]: Updating migrations will not work for people who have already run them. Can you create a new migration that changes the fields?

**Review by bikalbasnet — COMMENTED** [review]
Non blocker comments. can merge.

File: backend/routes/api.php [hunk 6]
```diff
@@ -29,17 +29,29 @@
 Route::post('/auth', 'AuthController@auth');
 Route::get('/auth/test', 'AuthController@test')->middleware('auth:api');
 
+
+/*
+ * Design
+ */
+Route::prefix('design')->group(function () {
+    Route::get('/', 'DesignController@index')->middleware('jsonapi');
```
Comment (AlexStansfield) [inline backend/routes/api.php]: I don't think this middleware is needed

**Review by AlexStansfield — COMMENTED** [review]
Comments

**Pull request merged**